src/PerfView/OtherSources/Linux/LinuxPerfScriptEventParser.cs

```diff
@@ -346,9 +346,13 @@ private IEnumerable<LinuxEvent> NextEvent(Regex regex, FastStream source)
 
                 // CPU
                 source.SkipWhiteSpace();
-                source.MoveNext(); // Move past the "["
-                int cpu = source.ReadInt();
-                source.MoveNext(); // Move past the "]"
+                int cpu = -1;
+                if (source.PeekString(1) == "[")
```

Comment:
Maybe there is a better way so that this is only checked once, instead of on every cpu-clock event.  I assume if one doesn't have cpu number, the rest won't have it either.